pkg/util/net.go

```diff
@@ -142,3 +146,37 @@ func addNoProxyEnvVarEntries(entries map[string]struct{}, envVar string) {
 		}
 	}
 }
+
+func selectV4IPFromHostInterface() (string, error) {
+	ifaces, err := tcpnet.Interfaces()
+	if err != nil {
+		return "", err
+	}
+
+	// get list of interfaces
```

Comment:
Are these lists sorted in anyway? The loops return the first time they find any IP at all, so a host that has several might get different values at different times if the list of interfaces and then addresses on the interface are not sorted. I don't know which direction we would want to sort, but we do want the results to be deterministic.

Reply:
Yes, I agree. Since we are implementing a fallback in case of non-existing default route, we may only want to use it in case we have one and only IPv4 address. Otherwise, we're risking non-deterministic behavior.

Reply:
This list is sorted by the index of interfaces, from little to big. The result is deterministic if no new netdevice is added.